scripts/typescript/src/bcdb/bcdb.ts

```diff
@@ -0,0 +1,352 @@
+          if (!reference) {
+            reference = {
+              node: ["Reference"],
+              doi: doi,
+              author: [author], // TODO: parse array
+              type: "", // TODO: pick a reference-type key from vocab
+            };
+            if (reference.doi) {
+              references.set(reference.doi, reference);
+            } else {
+              error(index, `Unable to store this reference, it has no doi (we use it as a key)`);
+            }
+          }
+          //log(`Reference read: ${reference.doi}`);
+
+          //-- citation
+          //   to reference the reference (haha)
+          const citation: ICitation = {
+            // uid: will be determined by CriptJSON serializer
+            name: `RCBC Citation ${index}`,
+            node: ["Citation"],
+            type: "reference",
+            reference,
+          };
+
+          //-- product (Overall polymer)
+          //   note: we do not reuse existing polymers,
+          //         because we have to attach similar properties
+          //         on similar polymers (matching in term of bigsmiles)
+          const bigsmiles = row[Column.BigSMILES];
+          const polymer: IMaterial = {
+            node: ["Material"],
+            name: `RCBC Material ${index}`,
+            bigsmiles,
+            notes: row[Column.notes]
+          };
+          project.material.push(polymer);
+
+          // log(`Added polymer: ${polymer.bigsmiles}`);
+
+          //-- product's properties
+          //   mw_n, mw_w, mw_d, invariant_degree_of_polymerization
+          if (!polymer.property) polymer.property = [];
+          polymer.property.push({
+            node: ["Property"],
+            key: "mw_n",
+            value: row[Column.Mn],
+            method: row[Column.Mn_method],
+            unit: "g/mol",
+          });
+          polymer.property.push({
+            node: ["Property"],
+            key: "mw_w",
+            value: row[Column.Mw],
+            method: row[Column.Mw_method],
+            unit: "g/mol",
+          });
+          polymer.property.push({
+            node: ["Property"],
+            key: "mw_d",
+            value: row[Column.D],
+            method: row[Column.D_method],
+            unit: "g/mol",
+          });
+          polymer.property.push({
+            node: ["Property"],
+            key: "invariant_degree_of_polymerization",
+            value: row[Column.N],
+            method: row[Column.N_method],
+            //unit: undefined
+          });
+          polymer.property.push({
+            node: ["Property"],
+            key: "temperature",
+            unit: "degC",
+            value: row[Column.T],
+          });
+
+          const phase_method = row[Column.phase_method] ?? "";
+          const condition: ICondition = {
+            node: ["Condition"],
+            key: "phase_method",
+            value: phase_method.toLowerCase(), // saxs, tem, rheology
```

Comment:
`rheology` is not available in the vocab. We have to add it or pick a different one.